source/Core/Src/gui.cpp

```diff
@@ -894,6 +902,32 @@ static void settings_displayPowerPulse(void) {
     OLED::print(OffString);
   }
 }
+
+static bool settings_setAnimationLoop(void) {
+  systemSettings.animationLoop = !systemSettings.animationLoop;
+  return false;
+}
+
+static void settings_displayAnimationLoop(void) {
+  printShortDescription(29, 7);
+  OLED::drawCheckbox(systemSettings.animationLoop);
+}
+
+static bool settings_setAnimationSpeed(void) {
+  systemSettings.animationSpeed += 50;
```

Comment:
Rather than using magic numbers here for the times, can you use the `TICKS_` constants
Such as `TICKS_100MS`, `TICKS_SECOND` (and you can divide / multiply these to get what you need.